anvil/src/shell/x11.rs

```diff
@@ -241,13 +243,38 @@ impl<BackendData: Backend> XwmHandler for CalloopData<BackendData> {
         self.state.move_request_x11(&window)
     }
 
+    fn allow_selection_access(&mut self, xwm: XwmId, _selection: SelectionType) -> bool {
+        if let Some(keyboard) = self.state.seat.get_keyboard() {
+            // check that an X11 window is focused
+            if let Some(FocusTarget::Window(WindowElement::X11(surface))) = keyboard.current_focus() {
+                if surface.xwm_id().unwrap() == xwm {
+                    return true;
+                }
+            }
+        }
+        false
+    }
+
+    fn send_selection(&mut self, _xwm: XwmId, selection: SelectionType, mime_type: String, fd: OwnedFd) {
+        match selection {
+            SelectionType::Clipboard => {
+                if let Err(err) = request_data_device_client_selection(&self.state.seat, mime_type, fd) {
+                    error!(?err, "Failed to request current wayland clipboard for Xwayland",);
+                }
+            }
+            SelectionType::Primary => {
+                if let Err(err) = request_primary_client_selection(&self.state.seat, mime_type, fd) {
+                    error!(
+                        ?err,
+                        "Failed to request current wayland primary selection for Xwayland",
+                    );
+                }
+            }
+        }
+    }
+
     fn new_selection(&mut self, _xwm: XwmId, selection: SelectionType, mime_types: Vec<String>) {
-        slog::info!(
-            self.state.log,
-            "Got Selection {:?} from X11: {:?}",
-            selection,
-            mime_types
-        );
+        info!(?selection, ?mime_types, "Got Selection from X11",);
```

Comment:
I'm confused as to why this is in this PR...?

Reply:
I think because the tracing migration was midway and during the rebase I merged this fix with the wrong commit. :sweat_smile: 
Not super harmful, but I can clean that up.